LayoutKit/Layouts/SizeLayout.swift

```diff
@@ -88,28 +151,90 @@ public class SizeLayout<V: View>: BaseLayout<V>, Layout {
                   config: config)
     }
 
-    public func measurement(within maxSize: CGSize) -> LayoutMeasurement {
-        let size = CGSize(width: width ?? .max, height: height ?? .max)
-        var constrainedSize = size.sizeDecreasedToSize(maxSize)
-
-        // If at least one dimension is nil, then we need to measure the sublayout to inherit its value (zero if there is no sublayout).
-        if width == nil || height == nil {
-            let subsize = sublayout?.measurement(within: constrainedSize).size ?? CGSizeZero
-            if width == nil {
-                constrainedSize.width = subsize.width
-            }
-            if height == nil {
-                constrainedSize.height = subsize.height
-            }
+    public convenience init(maxSize: CGSize,
```

Comment:
Actually, we might move toward making our primitive layouts unsubclassable (encourage composition over inheritance) so my point about subclassing might be irrelevant.